### PR TITLE
15618 images template

### LINKED ISSUE #15618: Finalize Images template FE - mobile and desktop - learning center 1.1
## Finalize Images template FE - mobile and desktop - learning center 1.1

This is a follow on from the pre-work we already finished during 1.0: 
https://app.zenhub.com/workspaces/vft-59c95ae5fda7577a9b3184f8/issues/department-of-veterans-affairs/va.gov-team/13499

Test node with Images article content: http://preview-prod.vfs.va.gov/preview?nodeId=6954

AC:
- [x]  The FE is accessibility compliant
- [x] The FE works as mobile responsive view
- [ ] ~Pagination~ Show more feature is added per design ( `10 images per page`) - 12/9 updated
- [ ]  Design implementation is validated by @rtwell 

**Images article content fields - required unless noted as optional:**

H1
Intro text
Info alert `(optional)`
CTA button(s)
H2 section header` (optional)`   **_<-- Currently not showing in the FE though it is there in the CMS_**
Image
Image title
Image description `(optional)`  **_<-- Currently missing in CMS_**
Pagination if more than 10 images on a page

### PR DESCRIPTION
**Pull request by youngfreezyVA**
PR for images template, with additional info component in lieu of pagination
resolves https://github.com/department-of-veterans-affairs/va.gov-team/issues/15618

### CODE REVIEW

**Review by kelsonic — APPROVED** [review]
I think the formatting of the liquid template might be a bit off (4 spaces instead of 2) but other than that lgtm! 🎉

File: src/site/includes/image_detail_mobile_responsive.drupal.liquid [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+                <picture>
+                    {%if mediaImage.entity.image.url == empty %}
+                        <img class="vads-u-min-height--viewport vads-u-background-color--gray-lightest vads-u-display--block vads-u-border--1px vads-u-border-color--gray-light" alt="{{ alt }}" height="{{ mediaImage.entity.image.height }}" width="{{ mediaImage.entity.image.width }}" />
+                    {% else %}
+                        <img class="vads-u-border--1px vads-u-border-color--gray-light" alt="{{ alt }}" height="{{ mediaImage.entity.image.height }}" width="{{ mediaImage.entity.image.width }}" src="{{ mediaImage.entity.image.url }}" />
+                    {%endif %}    
+                </picture>
+            </dd>
+            <dt class=".vads-l-col--12 vads-l-col--10 medium-screen:vads-u-padding-left--2 medium-screen:vads-u-margin-top--0">
+                <{{image_title_tag}} class="{{image_title_classes }} vads-u-display--none  medium-screen:vads-u-display--block medium-screen:vads-u-margin-y--0">{{ mediaImage.entity.entityLabel }}</{{image_title_tag}}>
```
Comment (kelsonic) [inline src/site/includes/image_detail_mobile_responsive.drupal.liquid:18]: Wow that's crazy that liquid supports that syntax lol

File: src/site/includes/image_detail_mobile_responsive.drupal.liquid [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    {% assign alt = mediaImage.entity.image.alt %}
+    <div class="vads-l-grid-container vads-u-padding-x--0 vads-u-margin-y--2 ">
+        <div class="vads-l-row">
+            <dt class="vads-l-col--12 medium-screen:vads-u-display--none vads-u-margin-bottom--2">
+                <{{image_title_tag}} class="{{image_title_classes }} medium-screen:vads-u-margin-top--0">{{ mediaImage.entity.entityLabel }}</{{image_title_tag}}>
+            </dt>
+            <dd class=".vads-l-col--12 medium-screen:vads-l-col--2">
+                <picture>
+                    {%if mediaImage.entity.image.url == empty %}
+                        <img class="vads-u-min-height--viewport vads-u-background-color--gray-lightest vads-u-display--block vads-u-border--1px vads-u-border-color--gray-light" alt="{{ alt }}" height="{{ mediaImage.entity.image.height }}" width="{{ mediaImage.entity.image.width }}" />
```
Comment (kelsonic) [inline src/site/includes/image_detail_mobile_responsive.drupal.liquid:11]: So is this just a blank image then?

File: src/site/includes/images_list_with_additional_info_component.drupal.liquid [hunk 3]
```diff
@@ -0,0 +1,26 @@
+<!-- this template needs to be included inside a <dl/> -->
+<!-- media list -->
+{%if images.size != 0%} 
+    {% assign pageSize = 10 %}
+    {% assign remainingImages = images | sliceArrayFromStart: pageSize, images.size %}
+    {% for mediaImage in images %}
+        {% assign recordsCount = recordsCount | plus: 1 %}
+        {%unless recordsCount > pageSize %}
+        {% include "src/site/includes/image_detail_mobile_responsive.drupal.liquid"
```
Comment (kelsonic) [inline src/site/includes/images_list_with_additional_info_component.drupal.liquid:9]: I know this is probably auto-formatted but could we indent this?

File: src/site/layouts/media_list_images.drupal.liquid [hunk 4]
```diff
@@ -4,109 +4,92 @@
...
+                            {% assign image_title_tag = "h3" %}
+                            {% assign image_title_classes = "vads-u-margin-y--0 medium-screen:vads-u-margin-y--1p5" %}
+                            <h2 class="vads-u-margin-bottom--4">{{fieldMediaListImages.entity.fieldSectionHeader }}</h2>
+                        {% else %}
+                            {% assign image_title_tag = "h2" %}
+                            {% assign image_title_classes = "vads-u-font-size--h3" %}
+                        {% endif %}
+                        <!-- Images List -->
+  
+                        <dl class="usa-unstyled-list">
```
Comment (kelsonic) [inline src/site/layouts/media_list_images.drupal.liquid:55]: I think the images list comment should be moved down one line right?

File: src/site/layouts/media_list_images.drupal.liquid [hunk 5]
```diff
@@ -4,109 +4,92 @@
...
 
-          <!-- VA benefits -->
-          {% include "src/site/includes/benefit-hubs-links.drupal.liquid" with fieldRelatedBenefitHubs = fieldRelatedBenefitHubs %}
+                    <!-- VA benefits -->
+                    {% include "src/site/includes/benefit-hubs-links.drupal.liquid" with fieldRelatedBenefitHubs = fieldRelatedBenefitHubs %}
 
-          <!-- Need more help -->
-          {% include "src/site/paragraphs/contact_information.drupal.liquid" with entity = fieldContactInformation.entity %}
+                    <!-- Need more help -->
+                    {% include "src/site/paragraphs/contact_information.drupal.liquid" with entity = fieldContactInformation.entity %}
```
Comment (kelsonic) [inline src/site/layouts/media_list_images.drupal.liquid:82]: Just so you know, I believe all `include`d templates have access to parent templates' variables by default 🙂

Comment (mpelzsherman) [reply src/site/layouts/media_list_images.drupal.liquid:82]: That's correct @kelsonic.

File: src/site/includes/image_detail_mobile_responsive.drupal.liquid (hunk 2 above)
Comment (youngfreezyVA) [reply src/site/includes/image_detail_mobile_responsive.drupal.liquid:11]: yup! this is for displaying the grey background when the alt text is displayed

File: src/site/layouts/media_list_images.drupal.liquid (hunk 4 above)
Comment (youngfreezyVA) [reply src/site/layouts/media_list_images.drupal.liquid:55]: yuppp

File: src/site/layouts/media_list_images.drupal.liquid (hunk 5 above)
Comment (youngfreezyVA) [reply src/site/layouts/media_list_images.drupal.liquid:82]: oh very cool, that's good to know. i'll remove the "prop-drilling" and see if i can get that working.

**Review by mdewey — APPROVED** [review]

File: src/site/includes/image_detail_mobile_responsive.drupal.liquid [hunk 6]
```diff
@@ -0,0 +1,20 @@
+{% assign alt = mediaImage.entity.image.alt %}
+<div class="vads-u-display--flex vads-u-flex-direction--column  medium-screen:vads-u-margin-bottom--2 vads-u-margin-bottom--4">
+    <dt class="vads-u-margin-bottom--2 medium-screen:vads-u-margin-bottom--0">
+        <{{image_title_tag}} class="{{image_title_classes }} medium-screen:vads-u-margin-top--0">{{ mediaImage.entity.entityLabel }}</{{image_title_tag}}>
```
Comment (billfienberg) [inline src/site/includes/image_detail_mobile_responsive.drupal.liquid:4]: `image_title_tag` is declared here:
https://github.com/department-of-veterans-affairs/vets-website/pull/15270/files#diff-82e7af6315a8edf6ddfd71183d365b82c51fc78e99cde418334bad49c8191b04R46

File: src/site/includes/images_list_with_additional_info_component.drupal.liquid [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+                with mediaImage = mediaImage  %}
+            {% endunless %}
+        {% endfor %}
+    {% if recordsCount > pageSize %}
+        <div class="form-expanding-group additional-info-container vads-u-margin-y--2p5">
+            <span class="additional-info-title">Show more</span>
+            <span>
+                <div class="additional-info-content">
+                {% for remainingImage in remainingImages %}
+                    {% include "src/site/includes/image_detail_mobile_responsive.drupal.liquid"
```
Comment (billfienberg) [inline src/site/includes/images_list_with_additional_info_component.drupal.liquid:19]: Leaving this here for myself/other reviews

Link to `src/site/includes/image_detail_mobile_responsive.drupal.liquid`: https://github.com/department-of-veterans-affairs/vets-website/pull/15270/files#diff-45a75abadb6bef1c40c2e4202e5f5e9d4155156b97c3563185bb994a055463dfR1

**Review by billfienberg — APPROVED** [review]
Ideally, we would fix the root cause behind a failing test instead of skipping it.

But since CircleCI is currently non-blocking, Jenkins passes, and the code makes sense, I'll approve the PR.

**Pull request merged**